crates/assembly/src/assembler.rs

```diff
@@ -969,20 +949,17 @@ impl Assembler {
                         split_builder.append_before_enter(decorator_ids);
                     }
 
-                    // Add an assembly operation decorator to the if node in debug mode.
-                    if self.in_debug_mode() {
-                        let location = proc_ctx.source_manager().location(*span).ok();
-                        let context_name = proc_ctx.path().to_string();
-                        let num_cycles = 0;
-                        let op = "if.true".to_string();
-                        let should_break = false;
-                        let op =
-                            AssemblyOp::new(location, context_name, num_cycles, op, should_break);
-                        let decorator_id = block_builder
-                            .mast_forest_builder_mut()
-                            .ensure_decorator(Decorator::AsmOp(op))?;
-                        split_builder.append_before_enter([decorator_id]);
-                    }
+                    // Add an assembly operation decorator to the if node.
+                    let location = proc_ctx.source_manager().location(*span).ok();
+                    let context_name = proc_ctx.path().to_string();
+                    let num_cycles = 0;
+                    let op = "if.true".to_string();
+                    let should_break = false;
+                    let op = AssemblyOp::new(location, context_name, num_cycles, op, should_break);
+                    let decorator_id = block_builder
+                        .mast_forest_builder_mut()
+                        .ensure_decorator(Decorator::AsmOp(op))?;
+                    split_builder.append_before_enter([decorator_id]);
```

Comment:
Not from this PR, but this code seems quite similar to the code on lines 1020-1030 below. Would it make sense to encapsulate this (or at least a part of this) in a helper function?